evals/elsuite/basic/match.py

```diff
@@ -36,18 +35,16 @@ def eval_sample(self, sample: Any, *_):
                 prompt += s["sample"]
             prompt += sample["input"][-1:]
 
-        # TODO(hwc): is there a case where we want to use `result` other than "choices"?
-        result = self._completion_fn(
+        result = self.completion_fn(
             prompt=prompt,
             temperature=0.0,
```

Comment:
Some completionfns won't have a temperature arg. Should we remove these references to temperature?

Reply:
Also, we should consider that with current design, every implementation of CompletionFn would need to implement a way to handle both Chat and non-chat prompts

Reply:
Had to do this in a test example:
```
from evals.prompt.base import CompletionPrompt

class MyCompletionResult():
    def __init__(self, raw_data) -> None:
        self.raw_data = raw_data

    def get_completions(self) -> list[str]:
        return [self.raw_data]

class MyCompletionFn():
    def __call__(self, prompt: str) -> MyCompletionResult:
        completion_prompt = CompletionPrompt(raw_prompt=prompt)
        return MyCompletionResult(completion_prompt.to_openai_create_prompt())
```

Reply:
We might want to rename`base.py:CompletionPrompt.to_openai_create_prompt` if it is going to be used to generically render chat prompts into text formats

Reply:
> Some completionfns won't have a temperature arg. Should we remove these references to temperature?

Good point, we should remove the reference here since it's specific to OpenAI models. However, I still want this eval to use zero temperature sampling for OpenAI models in this type of eval where there's a small set of correct answers since we want to have the model output the most likely answer. I brainstormed a few options but wasn't too satisfied with any of them. Any thoughts?

> Also, we should consider that with current design, every implementation of CompletionFn would need to implement a way to handle both Chat and non-chat prompts

Yep. Should we simplify the CompletionFn interface to have prompt be either `list[dict]` (for chat) or `str` (all others)? Right now the prompt type is very specific to the OpenAI API.

Reply:
How about putting the temperature 0 inside of the completionfn?

> Yep. Should we simplify the CompletionFn interface to have prompt be either list[dict] (for chat) or str (all others)? Right now the prompt type is very specific to the OpenAI API.

The underlying implementation is pretty much just list[dict] or str. I think we would just need to rename some things.
```
OpenAICreatePrompt = Union[str, list[str], list[int], list[list[int]]]

# This is the type accepted as the `prompt` field to `openai.ChatCompletion.create` calls
OpenAIChatMessage = Dict[str, str]  # A message is a dictionary with "role" and "content" keys
OpenAICreateChatPrompt = List[OpenAIChatMessage]  # A chat log is a list of messages
```